crates/sc-proof-of-time/src/source/gossip.rs

```diff
@@ -265,25 +265,24 @@ where
                     "Proof from the future",
                 );
 
-                let proofs = self
-                    .gossip_cache
-                    .get_or_insert_mut(sender, Default::default);
-                if proofs.len() == GOSSIP_CACHE_PER_PEER_SIZE {
-                    if let Some(proof) = proofs.pop_front() {
-                        trace!(
-                            %sender,
-                            slot = %proof.slot,
-                            next_slot = %next_slot_input.slot,
-                            "Too many proofs stored from peer",
-                        );
-
-                        self.engine
-                            .lock()
-                            .report(sender, rep::GOSSIP_TOO_MANY_PROOFS);
+                if let Some(proofs) = self.gossip_cache.get_or_insert(sender, Default::default) {
```

Comment:
Not sure why it is necessary, asked a question about this API upstream: https://github.com/koute/schnellru/issues/7